src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -131,6 +163,30 @@ public ReadPileup getPileupForLane(final String laneID) {
     }
 
     /**
+     * Splits the ReadPileup by sample
+     *
+     * @param header              the header to retrieve the samples from
+     * @param unknownSampleName the sample name if there is no read group/sample name; {@code null} if lack of RG is not expected
+     * @return a Map of sample name to ReadPileup (including empty pileups for a sample)
+     * @throws org.broadinstitute.hellbender.exceptions.UserException.ReadMissingReadGroup if unknownSampleName is {@code null} and there are reads without RG/sample name
+     */
+    public Map<String, ReadPileup> splitBySample(final SAMFileHeader header, final String unknownSampleName) {
+        final Map<String, ReadPileup> toReturn = new HashMap<>();
+        for (final String sample : getSamples(header)) {
+            final ReadPileup pileupBySample = getPileupForSample(sample, header);
+            if (sample != null) {
+                toReturn.put(sample, pileupBySample);
+            } else {
+                if (unknownSampleName == null) {
+                    throw new UserException.ReadMissingReadGroup(pileupBySample.iterator().next().getRead());
```

Comment:
please add a test case that reaches this line `throw new UserException.ReadMissingReadGroup(pileupBySample.iterator().next().getRead());`